### DISCUSSION TITLE
"Unknown request to /identity"

### DISCUSSION THREAD

**Discussion by ghost** [tgorgdotcom/locast2plex#211] [Q&A] [answered]
I have locast2plex installed on a Raspberry pi 4.  Running it via command line.  Started up fine and finished with "Locast2Plex is now online."

When I go into Plex on my Roku and manually add the IP and port, it seems to connect but in the console on the pi I get
"192.168.x.x - - [02/Feb/2021 11:10:45] "GET /identity HTTP/1.1 501 -"

Thoughts?

**Comment by CTJohnK** [comment]
Where is Plex installed and running?  I had L2P running on my Pi 4 for a long time without issue.

Can you provide a bit more of the log (keeping any private data out of the listing)?

**Reply by ghost** [reply]
Hi.  Thanks.  Below is what I see in the console

```
pi@raspberrypi:~/locast2plex $ python3 main.py
Initiating Locast2Plex v0.6.5
Opening and Verifying Configuration File.
/home/pi/locast2plex/config/config.ini
/home/pi/locast2plex/config.ini
Loading Configuration File: /home/pi/locast2plex/config.ini
Tuner count set to 3
Server is set to run on  192.168.2.206:6077
No UUID found.  Generating one now...
UUID set to: txhvnvkw...
Getting user location...
Getting location via IP Address.
Got external IP xx.xx.xx.xx.
Got location as Boston - DMA 506 - Lat\Lon 42.3584\-71.0598
Logging into Locast using username my@email.com...
Validating User Info...
User Info obtained.
User didDonate: True
User donationExpire: 2021-02-27 19:37:33
Starting First time Stations refresh...
Checking FCC facilities database for updates.
Online facilities database is newer. Downloading the latest FCC facilities database...
Unzipping FCC facilities database...
Reading and formatting FCC database...
Found 6387 stations.
Getting list of stations based on DMA...
Found 38 stations for DMA 506
Starting Stations thread...
Starting device server on 192.168.2.206:6077
Now listening for requests.
Starting SSDP server...
Starting First time EPG refresh...
Registering upnp:rootdevice (http://192.168.2.206:6077/device.xml)
Fetching:   https://api.locastnet.org/api/watch/epg/506?startTime=2021-02-02T00:00:00
Fetching:   https://api.locastnet.org/api/watch/epg/506?startTime=2021-02-03T00:00:00
Fetching:   https://api.locastnet.org/api/watch/epg/506?startTime=2021-02-04T00:00:00
Fetching:   https://api.locastnet.org/api/watch/epg/506?startTime=2021-02-05T00:00:00
Fetching:   https://api.locastnet.org/api/watch/epg/506?startTime=2021-02-06T00:00:00
Fetching:   https://api.locastnet.org/api/watch/epg/506?startTime=2021-02-07T00:00:00
Starting EPG thread...
Locast2Plex is now online.
192.168.2.23 - - [02/Feb/2021 11:08:46] "GET /identity HTTP/1.1" 501 -
Unknown request to /identity
192.168.2.23 - - [02/Feb/2021 11:10:45] "GET /identity HTTP/1.1" 501 -
```

**Reply by ghost** [reply]
Additional info... if from my raspberry pi browser I hit 192.168.2.206:6077 I get the following (which appears to be meaningful)

```<MediaContainer>
<MediaGrabber identifier="tv.plex.grabbers.locast2plex" title="Locast2Plex" protocols="livetv"/>
</MediaContainer>```

**Reply by ghost** [reply]
Additional info... locast2plex seems to be running just fine.  Overnight it refreshed its caches.  But again when I hit it this morning from my Roku Plex app I get `Unknown request to /identity`

```Fetching EPG for DMA 506.
Removing stale cache file: 02-02-2021.json
FROM CACHE: /home/pi/locast2plex/cache/506_epg/02-03-2021.json
FROM CACHE: /home/pi/locast2plex/cache/506_epg/02-04-2021.json
FROM CACHE: /home/pi/locast2plex/cache/506_epg/02-05-2021.json
FROM CACHE: /home/pi/locast2plex/cache/506_epg/02-06-2021.json
FROM CACHE: /home/pi/locast2plex/cache/506_epg/02-07-2021.json
Fetching:   https://api.locastnet.org/api/watch/epg/506?startTime=2021-02-08T00:00:00
```

**Reply by ghost** [reply]
Additional info... Just to make sure I am setting up the Roku Plex app here is what I do.  In Plex, I go to "Settings" and then click on "Manual Connections".  On this screen I add the IP and Port (6077) of my locast2plex server.  I assume this is correct.

**Comment by mschmittfl** [comment]
I am not at my Roku screen at the moment, however I have seen where the
Roku app does not have all the "admin" settings that are avail in the web
browser interface. I set mine up by using the browser and going to
settings, then under the 'Manage" section go to "Live TC & DVR" in this
menu "Add Device". The locast2plex server should show up here, if not there
is a dropdown to add it manually.

[image: image.png]

Once it is added you have the option to manage/verify the channels. After
the guide refreshes you should be able to watch via "Live TV & DVR" on the
home screen.

As I said, I am not looking at the Roku interface at the moment, but I have
seen where it seems to be more of a "user" interface/client.

**Reply by ghost** [reply]
Thanks for the reply mschmittfl.  I logged into the web interface for my Plex account.  I do not see a "Manage".  I have searched around and do not find anything like "Manage".  :-(
![image](https://user-images.githubusercontent.com/55995379/106748732-a08d3880-65f3-11eb-9f16-55815cecd513.png)

**Reply by ghost** [reply]
I am wondering... When I emailed Plex support the other day a person named "Kelvin" replied with the following

"The 'Ultimate Live TV and DVR' is a free feature available to all Plex users.  It is not a subscription."

I wonder if Kelvin was wrong.  If I go to Plex.tv and under "Premium" click on "Ultimate Live TV & DRV" the page that comes up indicates it costs $4.99 per month.  The reason I initially contacted Plex support was because I thought I needed "Ultimate Live TV & DVR" but when I purchased it, some how, I got the "Plex Pass".  When I tried to go back and purchase "Ultimate Live TV & DVR" instead, it would let me purchase it as I had already purchased "Plex Pass".  

I am really confused as to what I need to purchase, if anything, to get the DVR feature to work.

**Comment by mschmittfl** [comment]
Via the web browser interface (plexserverIP:32400) or http://app.plex.tv,
go to the settings page ( the crossed wrench and screwdriver icon in the
upper right)

From that page you will see:

[image: menu.png]

From there is where you can set up the Live TV & DVR (add locast2plex or
other Live stream devices)

**Comment by mschmittfl** [comment]
Yes you do need to "upgrade" to the 4.99 a month "Ultimate Live TV & DVR"
to get DVR capability

**Reply by ghost** [reply]
I guess I have a general question.  Maybe I was thinking incorrectly about how locast2plex works.  Do I need to have a Plex Media Server (PMS) between locast2plex and my Roku Plex App?
![locast2plex](https://user-images.githubusercontent.com/55995379/106824926-deb94500-6651-11eb-9af4-827c4d63dc81.png)

**Comment by mschmittfl** [comment, marked as answer]
Yes, The media server is where Media is stored and/or streamed from. The
Roku app is just a sort of client with some admin/customization
capabilities. The Plex Media Server is where you configure the main
parameters as to how media is stored, captured, streamed, presented, etc.

**Reply by ghost** [reply]
Thanks mschmittfl.  I guess that was a key part that I missed.  :-(. Thanks!
